components/rich_text_editor/rich_text_editor.vue

```diff
@@ -0,0 +1,250 @@
+<template>
+  <editor-content
+    :editor="editor"
+    data-qa="dt-rich-text-editor"
+  />
+</template>
+
+<script>
+import { Editor, EditorContent } from '@tiptap/vue-2';
+import Document from '@tiptap/extension-document';
+import Paragraph from '@tiptap/extension-paragraph';
+import Text from '@tiptap/extension-text';
+import {
+  RICH_TEXT_EDITOR_OUTPUT_FORMATS,
+  RICH_TEXT_EDITOR_AUTOFOCUS_TYPES,
+} from './rich_text_editor_constants';
+
+export default {
+  name: 'DtRichTextEditor',
+
+  components: {
+    EditorContent,
+  },
+
+  props: {
+    /**
+     * Value of the input
+     */
+    value: {
+      type: String,
+      default: '',
+    },
+
+    /**
+     * Whether the input is editable
+     */
+    editable: {
+      type: Boolean,
+      default: true,
+    },
+
+    /**
+     * Descriptive label for the input element
+     */
+    inputAriaLabel: {
+      type: String,
+      required: true,
+    },
+
+    /**
+     * Additional class name for the input element. Only accepts a String value
+     * because this is passed to the editor via options. For multiple classes,
+     * join them into one string, e.g. "d-p8 d-hmx96"
+     */
+    inputClass: {
+      type: String,
+      default: '',
+    },
+
+    /**
+     * Whether the input should receive focus after the component has been
+     * mounted. Either one of `start`, `end`, `all` or a Boolean or a Number.
+     * - `start`  Sets the focus to the beginning of the input
+     * - `end`    Sets the focus to the end of the input
+     * - `all`    Selects the whole contents of the input
+     * - `Number` Sets the focus to a specific position in the input
+     * - `true`   Defaults to `start`
+     * - `false`  Disables autofocus
+     * @values true, false, start, end, all, number
+     */
+    autofocus: {
+      type: [Boolean, String, Number],
+      default: false,
+      validator (autofocus) {
+        if (typeof autofocus === 'string') {
+          return RICH_TEXT_EDITOR_AUTOFOCUS_TYPES.includes(autofocus);
+        }
+        return true;
+      },
+    },
+
+    /**
+     * The output format that the editor uses when emitting the "@input" event.
+     * One of `text`, `json`, `html`. See https://tiptap.dev/guide/output for
+     * examples.
+     * @values text, json, html
+     */
+    outputFormat: {
+      type: String,
+      default: 'text',
+      validator (outputFormat) {
+        return RICH_TEXT_EDITOR_OUTPUT_FORMATS.includes(outputFormat);
+      },
+    },
+  },
+
+  emits: [
+    /**
+     * Editor input event
+     * @event input
+     * @type {String}
+     */
+    'input',
+
+    /**
+     * Event to sync the value with the parent
+     * @event input
+     * @type {String}
```

Comment:
Is this correct?

Reply:
Good point, when JSON is the output format it's not